aws/network/security_group_api_end_to_end_test.tf

```diff
@@ -0,0 +1,45 @@
+resource "aws_security_group" "api_end_to_end_test_lambda" {
+  description = "API end to end test Lambda"
+  name        = "api_end_to_end_test_lambda"
+  vpc_id      = aws_vpc.forms.id
+}
+
+resource "aws_security_group_rule" "local_lambda_to_idp_egress" {
+  description              = "Egress from Lambda to IdP for local communication"
+  type                     = "egress"
+  security_group_id        = aws_security_group.api_end_to_end_test_lambda.id
+  source_security_group_id = aws_security_group.idp_ecs.id
+  protocol                 = "tcp"
+  from_port                = 8080
+  to_port                  = 8080
+}
+
+resource "aws_security_group_rule" "local_lambda_to_idp_ingress" {
+  description              = "Ingress to IdP from Lambda for local communication"
+  type                     = "ingress"
+  security_group_id        = aws_security_group.idp_ecs.id
+  source_security_group_id = aws_security_group.api_end_to_end_test_lambda.id
+  protocol                 = "tcp"
+  from_port                = 8080
+  to_port                  = 8080
+}
+
+resource "aws_security_group_rule" "local_lambda_to_api_egress" {
+  description              = "Egress from Lambda to API for local communication"
+  type                     = "egress"
+  security_group_id        = aws_security_group.api_end_to_end_test_lambda.id
+  source_security_group_id = aws_security_group.api_ecs.id
+  protocol                 = "tcp"
+  from_port                = 3001
+  to_port                  = 3001
+}
+
+resource "aws_security_group_rule" "local_lambda_to_api_ingress" {
+  description              = "Ingress to API from Lambda for local communication"
+  type                     = "ingress"
+  security_group_id        = aws_security_group.api_ecs.id
+  source_security_group_id = aws_security_group.api_end_to_end_test_lambda.id
+  protocol                 = "tcp"
+  from_port                = 3001
+  to_port                  = 3001
+}
```

Comment:
Do you want to use the new rule resource types since there are no existing rules on this security group?
- aws_vpc_security_group_egress_rule
- aws_vpc_security_group_ingress_rule

Reply:
I was hesitating because I would need to mix both resource types since the IdP and API security groups are on the `aws_security_group_rule` one. Though I can do it and add a comment for the two different ones.
Unless I did not fully understand and the only reason we can't mix is when both resource types are declared in the same file no matter what the security group it is attached to.